godot-core/src/builtin/signal.rs

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use std::fmt;
+use std::ptr;
+
+use godot_ffi as sys;
+
+use super::meta::{impl_godot_as_self, FromGodot, GodotType};
+use crate::builtin::meta::ToGodot;
+use crate::builtin::{inner, Array, Callable, Dictionary, StringName, Variant};
+use crate::engine::{global::Error, Object};
+use crate::obj::bounds::DynMemory;
+use crate::obj::{Bounds, Gd, GodotClass, InstanceId};
+use sys::{ffi_methods, GodotFfi};
+
+/// A `Signal` represents a signal of an Object instance in Godot.
+///
+/// Signals are composed of a reference to an `Object` and the name of the signal on this object.
+#[repr(C, align(8))]
```

Comment:
Why the `align(8)`?

Reply:
Callable is also using `align(8)` and the Godot classes for both use `alignas(8)`, so I thought it was necessary.

https://github.com/godotengine/godot/blob/b7449a3d1aebd34026c7a2279a1b18ff0f02ccfa/core/variant/callable.h#L163-L171

Reply:
Makes sense!